.github/workflows/test.yml

```diff
@@ -169,3 +169,22 @@ jobs:
       - name: Test
         # WIP refactor, only tests in the specified dirs have been fixed
         run: GOARCH=386 go test ./parquet/file/...
+  tinygo:
+    name: TinyGo
+    runs-on: ubuntu-latest
+    env:
+      TINYGO_VERSION: 0.33.0
+    timeout-minutes: 15
+    steps:
+      - name: Checkout
+        uses: actions/checkout@692973e3d937129bcbf40652eb9f2f61becf3332 # v4.1.7
```

Comment:
is `v4.1.7` not a tagged verison we can use directly? Do we have to use the sha?

Reply:
We can use `v4.1.7` directly. This is for security.
See also: https://github.com/apache/arrow/pull/37676

(Honestly, I'm using tag directly in general...)

@assignUser may be able to explain security more.